RemoteClusterConnectionTests#testEnsureConnected fails

**Build scan**: https://gradle-enterprise.elastic.co/s/xecfifgq5vtte

```
 java.lang.AssertionError: ExecutionCancelledException[operation was cancelled reason [connect handler is closed]]
  2> 	at __randomizedtesting.SeedInfo.seed([CC290201858E81C0]:0)
  2> 	at org.elasticsearch.transport.RemoteClusterConnectionTests$11.onFailure(RemoteClusterConnectionTests.java:1181)
  2> 	at org.elasticsearch.action.LatchedActionListener.onFailure(LatchedActionListener.java:50)
  2> 	at org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:50)
  2> 	at org.elasticsearch.action.ActionListener.onFailure(ActionListener.java:140)
  2> 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler$1.lambda$doRun$1(RemoteClusterConnection.java:506)
  2> 	at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:70)
  2> 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler.collectRemoteNodes(RemoteClusterConnection.java:582)
  2> 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler.access$1000(RemoteClusterConnection.java:413)
  2> 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler$1.doRun(RemoteClusterConnection.java:511)
  2> 	at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)
  2> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  2> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  2> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:708)
  2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  2> 	at java.lang.Thread.run(Thread.java:748)
  2> Caused by: ExecutionCancelledException[operation was cancelled reason [connect handler is closed]]
  2> 	at org.elasticsearch.common.util.CancellableThreads.checkForCancel(CancellableThreads.java:65)
  2> 	at org.elasticsearch.common.util.CancellableThreads.executeIO(CancellableThreads.java:130)
  2> 	at org.elasticsearch.transport.RemoteClusterConnection$ConnectHandler.collectRemoteNodes(RemoteClusterConnection.java:523)
  2> 	... 9 more
```

--- a/server/src/test/java/org/elasticsearch/transport/RemoteClusterConnectionTests.java
+++ b/server/src/test/java/org/elasticsearch/transport/RemoteClusterConnectionTests.java
@@ -1170,7 +1170,7 @@ public class RemoteClusterConnectionTests extends ESTestCase {
                     assertFalse(connectionManager.nodeConnected(seedNode));
                     assertFalse(connectionManager.nodeConnected(discoverableNode));
                     assertTrue(connection.assertNoRunningConnections());
-                    CountDownLatch latch = new CountDownLatch(1);
+                    CountDownLatch firstLatch = new CountDownLatch(1);
                     connection.ensureConnected(new LatchedActionListener<>(new ActionListener<Void>() {
                         @Override
                         public void onResponse(Void aVoid) {
@@ -1180,12 +1180,13 @@ public class RemoteClusterConnectionTests extends ESTestCase {
                         public void onFailure(Exception e) {
                             throw new AssertionError(e);
                         }
-                    }, latch));
-                    latch.await();
+                    }, firstLatch));
+                    firstLatch.await();
                     assertTrue(connectionManager.nodeConnected(seedNode));
                     assertTrue(connectionManager.nodeConnected(discoverableNode));
                     assertTrue(connection.assertNoRunningConnections());
 
+                    CountDownLatch secondLatch = new CountDownLatch(1);
                     // exec again we are already connected
                     connection.ensureConnected(new LatchedActionListener<>(new ActionListener<Void>() {
                         @Override
@@ -1196,8 +1197,8 @@ public class RemoteClusterConnectionTests extends ESTestCase {
                         public void onFailure(Exception e) {
                             throw new AssertionError(e);
                         }
-                    }, latch));
-                    latch.await();
+                    }, secondLatch));
+                    secondLatch.await();
                     assertTrue(connectionManager.nodeConnected(seedNode));
                     assertTrue(connectionManager.nodeConnected(discoverableNode));
                     assertTrue(connection.assertNoRunningConnections());

